app/plugin.go

```diff
@@ -747,56 +760,65 @@ func (a *App) processPrepackagedPlugins(pluginsDir string) []*plugin.Prepackaged
 
 	pluginSignaturePathMap := getPluginsFromFilePaths(fileStorePaths)
 	plugins := make([]*plugin.PrepackagedPlugin, 0, len(pluginSignaturePathMap))
+	prepackagedPlugins := make(chan plugin.PrepackagedPluginResult, len(pluginSignaturePathMap))
 	for _, pluginPaths := range pluginSignaturePathMap {
-		plugin, err := a.processPrepackagedPlugin(pluginPaths)
-		if err != nil {
-			mlog.Error("Failed to install prepackaged plugin", mlog.String("path", pluginPaths.path), mlog.Err(err))
+		go a.processPrepackagedPlugin(pluginPaths, prepackagedPlugins)
```

Comment:
Instead of changing the signature of `processPrepackagedPlugin`, could we isolate the function here and just write the return values back to the channel ourselves? 

This would also allow us to inline `PrepackagedPluginResult` altogether, but I might also suggest avoiding the struct, logging the error directly, and only writing the plugin to the channel if no error occurs.  That in turn would require a little more coordination among the goroutines -- perhaps another `sync.WaitGroup` followed by a `close(prepackagedPlugins)`, allowing you range over the variable number of successfully installed plugins to append to the return value.

Reply:
@RajatVaryani, what are your thoughts on this further simplification?